### ISSUE TITLE
who retweeted me?

### ISSUE THREAD

**Issue by bunnymatic** [fmborghino/how-i-tweet#2]
hey there

how hard do you think it'd be to do who retweeted/favorited me?

You've already looked at the twitter gem/api so i thought you might have some sense.

Here's what our twitter owner said about the app

> Cool! I logged in to it. Do you know if this is showing people we've favorited the most? or people who favorite us?
> 
> If it's people who retweet and favorite us, that could be interesting to throw up on Carbometer as stats. I'm not sure what it tells us, the other way around? That we love Rudy's tweets? or just that he tweets a lot?:)
> 
> I'll tell the comms team about it when we have our meeting Friday.

**Comment by fmborghino** [comment]
I'll change the "menu" (such as it is) to clarify that this is "list what I favorited" etc.

As for the other way, there's a lot of tools already to tell you who loves YOU on twitter, but really not to remind you of what you loved in the past, which this addresses. Start typing [who favorited my tweets the most] into a search to see :)

However it would be pretty easy to add lists of (a) who mentioned me, (b) who retweeted me, (c) who favorited me overall (altho' generally it seems like it's more interesting to see who fav'd specific tweets, which is pretty much built in to any client).

I'll add (a) and (b) for a start.

Also need to drop in some off the shelf sensible UI, much as I like it the way it is, it's pretty useless on a small screen.

**Comment by bunnymatic** [comment]
to be clear - i'm just making suggestions. no need to comply.  though our
Twitter owner did find it pretty interesting.

She's also managing the 'socialness' of c5 and has found several other
tools that help figure out mentions and favorites and such.   i think (as
with many things) it's about having the one-stop shop as opposed to
compiling email from 3 different metric services that each have a little
bit different info.

for the UI, if you're thinking about nav and the rest, then maybe a drop in
might be good.  perhaps, this is the foray into angular... ?

I'll see if i can cobble an angular renderer together.  might be fun.

**Comment by fmborghino** [comment]
I was thinking a basic pure or bootstrap or whatever-it's-all-the-same to get rolling. Angular and D3 might be a hoot for simple numbers visualization and switching views. What do you guys like to start with? Want to hack for a couple of hours?

**Comment by bunnymatic** [comment]
yeah - i did the tiniest spike into angular.  i think it's more than you need.  bootstrap or foundation or even something like pure or oocss.  

yes we should hack.  let's plan something.

**Comment by fmborghino** [comment]
Leaving this issue open as a hook upon which to hang:

Add retweets-of-me view, sorted by user, same as the other views. Turns out this is a 2 step operation with the Twitter API: (1) Get a timeline of my retweeted tweets (maxes out at 100, which can be done in one GET) (2) Get the list of retweeters for each of those.

#2 requires a GET per tweet, which is going to hit rate limits really fast :/

It's time to plan a real rate-limit strategy and sensible caching.
